src/cnaas_nms/api/firmware.py

```diff
@@ -0,0 +1,112 @@
+import os
+import json
+import requests
+import yaml
+
+from flask import request
+from flask_restful import Resource
+from cnaas_nms.scheduler.scheduler import Scheduler
+from cnaas_nms.api.generic import empty_result
+from cnaas_nms.scheduler.wrapper import job_wrapper
+from cnaas_nms.tools.log import get_logger
+from cnaas_nms.tools.get_apidata import get_apidata
+
+logger = get_logger()
+
+
+apidata = get_apidata()
+if isinstance(apidata, dict) and 'httpd_url' in apidata:
+    HTTPD_URL = apidata['httpd_url']
+else:
+    HTTPD_URL = 'https://cnaas_httpd/api/v1.0/'
+
+
+@job_wrapper
+def get_firmware(**kwargs: dict) -> str:
+    try:
+        res = requests.post(HTTPD_URL, json=kwargs,
+                            verify=False)
+        json_data = json.loads(res.content)
+    except Exception as e:
+        logger.exception(f"Exception while getting firmware: {e}")
+        return 'Could not download firmware'
+    if json_data['status'] == 'error':
+        return json_data['message']
+    return 'File downloaded from: ' + kwargs['url']
+
+
+@job_wrapper
+def get_firmware_chksum(**kwargs: dict) -> str:
+    try:
+        url = HTTPD_URL + '/' + kwargs['filename']
+        res = requests.get(url, verify=False)
+        json_data = json.loads(res.content)
+    except Exception as e:
+        logger.exception(f"Exceptionb while getting checksum: {e}")
+        return 'Failed to get checksum for ' + kwargs['filename']
+    if json_data['status'] == 'error':
+        return json_data['message']
+    return json_data['data']['file']['sha1']
+
+
+@job_wrapper
+def remove_file(**kwargs: dict) -> str:
+    try:
+        url = HTTPD_URL + '/' + kwargs['filename']
+        res = requests.delete(url, verify=False)
```

Comment:
I'm not very good at this myself, but it's nice to have verify=False configurable. So for example in api.yml where you specify the URL to the httpd container, you could have an option to specify if the cert should be verified when accessing this url

Reply:
Wouldn't it be better to have this as an option to the API method? Then if the download fails due to an SSL error I as a user can override it.

Reply:
Hm, I guess it depends on why you want to do the override. My thinking was that in a lab environment you maybe don't care about valid certs so you disable the check for that entire setup, but in production you should always be able to verify the cert. But maybe there are cases in a production env where you want to disable the check as well?

Reply:
Guess you can end up in situations where you want to disable the check temporarily, but default should always be to validate. Configuring it in the config file will make people disable the validation and leave it like that forever, even in production.

Reply:
Haha, yes that might be true, but at least we did what we could! Anyway, I'm fine with either option, API option or config file option